internal/verifier/verifier.go

```diff
@@ -214,6 +221,16 @@ func (verifier *ProvenanceIRVerifier) Verify() (VerificationResult, error) {
 		combinedResult.Combine(nextResult)
 	}
 
+	// Verify BuilderImageDigest.
+	if verifier.Want.BuilderImageDigests != nil {
+		nextResult, err := verifier.Got.verifyBuilderImageDigest(verifier.Want)
+
+		if err != nil {
+			return combinedResult, fmt.Errorf("verify builder image digests failed: %v", err)
+		}
+		combinedResult.Combine(nextResult)
```

Comment:
How about using https://pkg.go.dev/go.uber.org/multierr instead of `VerificationResult`? (in a separate PR)

Reply:
Good idea: #184